Tegra warrior updates

dwalkes opened the pull request:
See details in [topmost commit](https://github.com/mendersoftware/meta-mender-community/pull/107/commits/ff4ebda38ce3337a5537d5624de780c911cfed4d) message

There are a lot of changes implemented with the warrior branch.  I thought about attempting to split this commit up and I could do this if desired.  The redundant boot components might be the most useful to split out.  I did include a [reference to the branch](https://github.com/Trellis-Logic/meta-mender-community/commits/warrior-before-rebase) which contains each individual commit.  Let me know if you'd like any reorganization done with this.

I've added a lot to the README.md which is warrior specific.  I think I'll need some subset of this duplicated on the mender hub page as well, or I could move any to a warrior specific section on the mender hub page.

Let me know if you have questions or requests for additional changes.

mirzak commented on meta-mender-tegra/recipes-mender/tegra-state-scripts/files/redundant-boot-reboot-state-script:
```diff
@@ -0,0 +1,29 @@
+    ln -s /opt/ota_package/bl_update_payload_current /opt/ota_package/bl_update_payload
+    exit 1
+fi
+exit 0
```
Nice!

mirzak commented on meta-mender-tegra/recipes-mender/tegra-state-scripts/files/LICENSE:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Do not think it is necessary to provide your own copy of the Apache 2.0 License, and instead you can use the following in your recipe:

```
LIC_FILE_CHECKSUM = "${COMMON_LICENSE_DIR}/Apache-2.0;md5=89aea4e17d99a7cacdbeed46a0096b10"
```

dwalkes replied:
OK agreed, will do, I was copying the example at https://github.com/mendersoftware/meta-mender/tree/master/meta-mender-demo/recipes-mender/example-state-scripts/files FYI.

mirzak commented on meta-mender-tegra/README.md:
```diff
@@ -17,7 +17,7 @@ Download the source:
             -u https://github.com/mendersoftware/meta-mender-community \
             -m meta-mender-tegra/scripts/manifest-tegra.xml \
-            -b thud
+            -b warrior
```
Can you also update the README structure to match the suggested structure that we discussed. You can use [this](https://github.com/mendersoftware/meta-mender-community/blob/warrior/meta-mender-raspberrypi/README.md) as a template

dwalkes replied:
Yes, sorry I missed that.

mirzak commented on meta-mender-tegra/README.md:
```diff
@@ -28,3 +28,20 @@ Build
 
+## Customizing Image Rootfs Size
+
+To customize the size reserved for your rootfs, add variable MENDER_STORAGE_TOTAL_SIZE_MB to your build/local.conf or image and define with a value which is twice the size of your desired rootfs in MB.
```
I do not know if this is needed here, and should be covered in docs.mender.io pretty well.

dwalkes replied:
Good point.  It didn't work right/well before warrior on tegra.  Probably better in a mender hub known issues section.

dwalkes replied:
I've moved to the updated known issues section on [mender hub](https://hub.mender.io/t/nvidia-tegra-jetson-tx2/123)

mirzak commented on meta-mender-tegra/README.md:
```diff
@@ -28,3 +28,20 @@ Build
+Both of these changes above require changes to the u-boot environment location, which are outlined in the mender hub conversation linked above and also 
+[this post](https://devtalk.nvidia.com/default/topic/1063652/jetson-tx2/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/) on the nvidia forum about my observation related to mmcblkboot1.  
+This means upgrade from sumo or thud would require changes to move u-boot environment location and most likely would also require redundant boot support implemented as a post update state script similar to what
+is done on the warrior branch but also including additional payload and tools from earlier branches to support redundant boot.  I have not actually attempted this yet myself.
```
Nice information, and definitely think that this should also be part of the MH post. I would suggest to put it in the "Known Issues" section which is part of the template of MH posts.

dwalkes replied:
Done, also added a section on standalone updates and bootloader components to introduce the mender-install-manual script

mirzak commented on meta-mender-tegra/recipes-core/images/tegra-minimal-initramfs.bbappend:
```diff
@@ -0,0 +1,5 @@
+# See https://hub.mender.io/t/warrior-tegra-build-cicular-dependencies/757/5 and comments from Mirza
+EXTRA_IMAGEDEPENDS_remove = "u-boot"
+# Don't need mender to work with this image
+IMAGE_FSTYPES_remove = "mender mender.bmap"
```
I think this specific problem has been fixed with this commit,

https://github.com/mendersoftware/meta-mender/commit/a944df86e0aa7e61385ea1867847f65b59f8aeb3

dwalkes replied:
Agreed, removed

mirzak commented on meta-mender-tegra/recipes-mender/tegra-state-scripts/files/redundant-boot-reboot-state-script:
```diff
@@ -0,0 +1,29 @@
+ln -s /opt/ota_package/bl_update_payload_new /opt/ota_package/bl_update_payload
+nv_update_engine --install 1>&2
+# Should never get here, nv_update_engine will reboot on its own after successful update
+rc=$?
```
If it it is never supposed to get there, I think one can drop the check of return code and simply fail by reaching the code. What do you think?

dwalkes replied:
I agree, that's a better solution

dwalkes commented:
@mirzak please take another look, I think I've implemented all of the requested changes.  I've also updated the [mender hub page](https://hub.mender.io/t/nvidia-tegra-jetson-tx2/123)
